### PR TITLE
Update doc

### PR DESCRIPTION
**Pull request by tbrezot**
Use README [template](https://www.notion.so/cosmian/README-Template-5195be27f5e34d258382a1fd0ebf5d56).

**Note**: No version bump, this will be added to the major release v4.0 from https://github.com/Cosmian/crypto_core/pull/17

### CODE REVIEW

**Review by Manuthor — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -1,47 +1,184 @@
...
+## Documentation
+
+The documentation can be generated using Cargo:
+```
+cargo docs
+```
 
-The default feature schemes can all be built to a WASM target.
+If is also available on
+[doc.rs](https://docs.rs/cosmian_crypto_core/latest/cosmian_crypto_core/).
```
Comment (ThibsG) [inline README.md:184]:
```suggestion
It is also available on [doc.rs](https://docs.rs/cosmian_crypto_core/latest/cosmian_crypto_core/).
```

File: README.md [hunk 2]
```diff
@@ -1,47 +1,184 @@
...
-This one of the fastest elliptic curves known when implementing these objects.
-Its security level is also 128 bits.
+Benchmarks can be run with:
+```
+cargo bench
+```
+
+## Features and Benchmarks
+
+The benchmarks given bellow are run on a Intel(R) Core(TM) i7-10750H CPU @ 3.20GHz.
```
Comment (ThibsG) [inline README.md:117]:
```suggestion
The benchmarks given below are run on a Intel(R) Core(TM) i7-10750H CPU @ 3.20GHz.
```

File: README.md [hunk 3]
```diff
@@ -1,47 +1,184 @@
...
+
+### Asymmetric Crypto
+
+This crate implements a Diffie-Hellman asymmetric key pair based on the
+Curve25519. This one of the fastest elliptic curves known at this time and its
```
Comment (ThibsG) [inline README.md:122]:
```suggestion
Curve25519. This is one of the fastest elliptic curves known at this time and it
```

File: README.md [hunk 4]
```diff
@@ -1,47 +1,184 @@
...
 
-This crate implements a Data Encapsulation Mechanism (DEM) based on the AES 256
-GCM algorithm, as described in the [ISO 2004](https://www.shoup.net/iso/std6.pdf).
-This implementation is 128-bits secure.
+The following example can be run using `cargo run --example demo`.
 
-It uses the [`aes_gcm`](https://docs.rs/aes-gcm/latest/aes_gcm/index.html)
-implementation of the AES GCM algorithm. This implementation makes use of the
-AES instruction set when available, which allows for a high encryption speed.
+```
```
Comment (ThibsG) [inline README.md:28]: I think in Markdown (at least with Github rendering), you can include source code file: https://github.com/marketplace/actions/markdown-embed-code-from-file#how-to-use

**Pull request merged**